bin/oneoff/data_fix/time_spent_milliseconds_to_seconds

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env ruby
+
+require_relative '../../../dashboard/config/environment'
+
+# The time_spent field in the user_levels table has been
+# recording time in milliseconds. This will cause time_spent
+# to max out at ~25 days. Instead we want to record time in
+# seconds. This will cause time_spent to max out at ~68 years
+#
+# Relevant entries include:
+# SELECT *
+# FROM user_levels
+# WHERE time_spent IS NOT NULL
+#   AND time_spent > 0;
+#
+# time_spent should be transformed as such
+# (time_spent.to_f/1000).ceil.to_i
+
+UserLevel.where("time_spent > ?", 0).where.not(time_spent: nil).each do |user_level|
```

Comment:
One way to mitigate the impact of running this query on such a big table would be to use a method like `find_each` ([info](https://guides.rubyonrails.org/active_record_querying.html#find-each)) that can be chained to a `where`, that gets the records in batches rather than trying to instantiate them all at once.

Reply:
Additionally, it might make sense to do this in a transaction so that if something goes wrong, you won't end up in a state where some records are in seconds and other are in milliseconds. A good example is probably t[his script](https://github.com/code-dot-org/code-dot-org/blob/staging/bin/oneoff/remove_password_for_code_studio_admin.rb) - although you would want all of your updates to happen in one transaction. That might negate the benefits of using `find_each`, though, depending on how transactions work.